neonvm-runner/cmd/main.go

```diff
@@ -487,17 +488,42 @@ func runQEMU(
 
 	callbacks = cpuServerCallbacks{
 		get: func(logger *zap.Logger) (*vmv1.MilliCPU, error) {
-			return lo.ToPtr(vmv1.MilliCPU(lastValue.Load())), nil
+			switch cfg.cpuScalingMode {
+			case vmv1.CpuScalingModeSysfs:
+				cpu, err := getNeonvmDaemonCPU()
+				if err != nil {
+					logger.Error("failed to get CPU from NeonVM Daemon", zap.Error(err))
+					return nil, err
+				}
+				storedCpu := vmv1.MilliCPU(lastValue.Load())
+				if storedCpu.RoundedUp() != cpu.RoundedUp() {
+					logger.Warn("CPU from NeonVM Daemon does not match stored value, returning daemon value to let controller reconcile correct state", zap.Any("stored", storedCpu), zap.Any("current", cpu))
+					return &cpu, nil
+				}
+				logger.Debug("Returning stored CPU value to controller", zap.Any("cpu", storedCpu))
+				return &storedCpu, nil
+
+			case vmv1.CpuScalingModeQMP:
+				return lo.ToPtr(vmv1.MilliCPU(lastValue.Load())), nil
+			default:
+				panic(fmt.Errorf("unknown CPU scaling mode %s", cfg.cpuScalingMode))
```

Comment:
nit: Let's use `%q` instead of `%s`? It makes it easier to identify unusual strings (e.g., empty string, etc.)

Reply:
done here and in other panics in the file